code/game/objects/items/handcuffs.dm

```diff
@@ -377,12 +377,10 @@
 	w_class = WEIGHT_CLASS_SMALL
 	breakouttime = 60
 
-/obj/item/restraints/legcuffs/bola/energy/throw_impact(atom/hit_atom, datum/thrownthing/throwingdatum)
-	if(iscarbon(hit_atom))
-		var/obj/item/restraints/legcuffs/beartrap/B = new /obj/item/restraints/legcuffs/beartrap/energy/cyborg(get_turf(hit_atom))
-		B.spring_trap(null, hit_atom)
-		qdel(src)
-	. = ..()
+/obj/item/restraints/legcuffs/bola/energy/ensnare(mob/living/carbon/C)
+	var/obj/item/restraints/legcuffs/beartrap/B = new /obj/item/restraints/legcuffs/beartrap/energy/cyborg(get_turf(C))
+	B.spring_trap(null, C)
+	qdel(src)
```

Comment:
you are passing atom here but expect carbon

Reply:
This code seems fine to me

Reply:
Ehh yeah, I think this is fine. The passed atom will never not be a carbon and it's logged alright.

Reply:
spring_trap takes in an atom anyway